lib/src/mock_document_reference.dart

```diff
@@ -119,20 +122,19 @@ class MockDocumentReference extends Mock implements DocumentReference {
     return Stream.value(
         MockDocumentSnapshot(this, _documentId, root, _exists()));
   }
+}
 
-  static Map<String, dynamic> _deepCopy(Map<String, dynamic> fromMap) {
-    final toMap = <String, dynamic>{};
-
-    fromMap.forEach((key, value) {
-      if (value is Map<String, dynamic>) {
-        toMap[key] = _deepCopy(value);
-      } else if (value is List) {
-        toMap[key] = List.from(value);
-      } else {
-        toMap[key] = value;
-      }
+/// Copies data replicating its inner Maps and Lists.
+dynamic deepCopy(dynamic fromData) {
```

Comment:
Maybe move this to util.dart?

Reply:
Moved this and validateDocumentValue to util.dart.